src/Core/Settings.h

```diff
@@ -901,6 +901,7 @@ class IColumn;
     M(Bool, restore_replace_external_engines_to_null, false, "Replace all the external table engines to Null on restore. Useful for testing purposes", 0) \
     M(Bool, restore_replace_external_table_functions_to_null, false, "Replace all table functions to Null on restore. Useful for testing purposes", 0) \
     M(Bool, create_if_not_exists, false, "Enable IF NOT EXISTS for CREATE statements by default", 0) \
+    M(Bool, enable_secure_identifiers, false, "If enabled, only secure identifiers are allowed", 0) \
```

Comment:
Please elaborate on what "secure" means. Also it makes sense to add this setting to documentation (settings.md)